packages/server/utils/isRequestToJoinDomainAllowed.ts

```diff
@@ -1,43 +1,73 @@
 import isCompanyDomain from './isCompanyDomain'
 import getRethink from '../database/rethinkDriver'
 import {RDatum} from '../database/stricterR'
+import isUserVerified from './isUserVerified'
+import User from '../database/types/User'
+import {DataLoaderWorker} from '../graphql/graphql'
+import isValid from '../graphql/isValid'
+import TeamMember from '../database/types/TeamMember'
 
 export const getEligibleOrgIdsByDomain = async (
   activeDomain: string,
-  viewerId: string,
-  limit?: number
+  userId: string,
+  dataLoader: DataLoaderWorker
 ) => {
-  const BIG_ENOUGH_LIMIT = 9999
-
   if (!isCompanyDomain(activeDomain)) {
     return []
   }
 
   const r = await getRethink()
 
-  return r
+  const orgs = await r
     .table('Organization')
     .getAll(activeDomain, {index: 'activeDomain'})
-    .filter((org: RDatum) => org('featureFlags').contains('promptToJoinOrg'))
-    .filter((org: RDatum) =>
-      r
+    .merge((org: RDatum) => ({
+      members: r
         .table('OrganizationUser')
         .getAll(org('id'), {index: 'orgId'})
-        .filter({inactive: false, removedAt: null})
+        .orderBy('joinedAt')
         .coerceTo('array')
-        .do((orgUsers: RDatum) =>
-          orgUsers
-            .count()
-            .gt(1)
-            .and(orgUsers.filter((ou: RDatum) => ou('userId').eq(viewerId)).isEmpty())
-        )
+    }))
+    .merge((org: RDatum) => ({
+      founder: org('members').nth(0).default(null),
+      billingLeads: org('members').filter({role: 'BILLING_LEADER', inactive: false}),
+      activeMembers: org('members').filter({inactive: false, removedAt: null}).count()
+    }))
+    .filter((org: RDatum) =>
+      org('activeMembers').gt(0).and(org('members').filter({userId}).isEmpty())
```

Comment:
Did we intentionally change this from 1 to 0 ?

Reply:
This was not intentional, I missed it. Was this just to exclude onboarding orgs? Because with this >= 2, there is a gap: how do I get my first user to join my org?

Reply:
@Dschoordsch yes, initially we wanted to exclude orgs with a single member. But I think it is fine like this

Reply:
I changed the logic back and adapted the tests